### PR TITLE
ci: add ShellCheck as a GH action

### PR DESCRIPTION
**Pull request by 7flying**
Adds a ShellCheck github action

### CODE REVIEW

File: .github/workflows/linters.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  pull_request:
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
```
Comment (travier) [inline .github/workflows/linters.yaml:17]: We generally don't add random GitHub actions to our projects. I'm working on adding one in https://github.com/coreos/repo-templates/pull/72 but it needs the checks to pass here first.

**Comment by travier** [comment]
First step would be to review and merge: https://github.com/coreos/coreos-installer-dracut/pull/11

File: .github/workflows/linters.yaml (hunk 1 above)
Comment (7flying) [reply .github/workflows/linters.yaml:17]: I see, will close this PR then

**Comment by travier** [comment]
Then we can enable it with https://github.com/coreos/repo-templates/pull/72

**Pull request closed without merging**